web/src/app/schedules/CalendarToolbar.js

```diff
@@ -80,6 +97,31 @@ export default function CalendarToolbar(props) {
   return (
     <Grid container spacing={2} className={classes.container}>
       <Grid item xs={12} lg={4} className={classes.primaryNavBtnGroup}>
+        <FilterContainer
```

Comment:
Hmm, it looks like this is because the calendar toolbar is rendered from within the calendar component and gets remounted on the new filter when it attempts to re-query. I think the only thing the toolbar gets from the calendar is the label, so we might be able to infer that ourselves based on the start date and render it outside of that loop.

Reply:
Not entirely sure how to best tackle this one- looks like we are dependent on more than just the title in the toolbar: date, label, onNavigate, onView, and view. `ScheduleCalendar` is still written as a class so it's proven a bit harder to work with, but I think ultimately we should render the entire toolbar outside of the calendar component potentially using portals. What do you think? If so, it'd probably be best to do so after merging as a fast follow-up (convert ScheduleCalendar to hooks, tsx, then fix filter closing bug).

Reply:
Definitely! Yeah I was thinking if we had to do a rewrite like this just to get around a finicky rendering bug, that would feel pretty bad once we want to start designing our own solution- so the idea of just making it fully work without those props is :100: imo! I'll create an issue for this one as well for scope.

Reply:
Issue created: #1489